base/parse-type.jl

```diff
@@ -0,0 +1,208 @@
+# Parses a string into a Julia type object, e.g. `Int`, `Array{Int, 2}`, etc.
```

Comment:
All the other files have a "this is a part of julia" header. Also this will be the only file in base/ with a hyphenated name. It would be more conventional to call it tryparse.jl.

```suggestion
# This file is a part of Julia. License is MIT: https://julialang.org/license
# Parses a string into a Julia type object, e.g. `Int`, `Array{Int, 2}`, etc.
```

Reply:
oops, thanks Lilith! I'm also happy to merge it back into parse.jl if you prefer.